Fix csv of problem responses silent failure

awaisdar001 opened the pull request:
## [EDUCATOR-4003](https://openedx.atlassian.net/browse/EDUCATOR-4003)

This fixes an issue that when a CAPA problem contained multiple inputs, the system does not format the input `label` with `markup` object instead of a string object. due to this, generating a CSV for problem responses for learners in instructor dashboard fails. 

In this PR, I made the behavior consistent with single input CAPA problem. For multiple input problems, create a markup object as well. 

### How to test it?
## Stage
* Answer the following Problem in LMS:
https://courses.stage.edx.org/courses/course-v1:ArbiRaees+CS101+2017_T2/courseware/98bc47d112114c7989c4f6f7759883cf/e1954f8e68914299b992b439dd0e994c/1
* Go to Instructor Dashboard > Data Download
https://courses.stage.edx.org/courses/course-v1:ArbiRaees+CS101+2017_T2/instructor#view-data_download
* Select a section or problem > Select Blank Advanced Problem (block-v1:ArbiRaees+CS101+2017_T2+type@problem+block@8cf9219027af41228180d911b192ced2)
* Click `Download a CSV of problem responses`
* Observe error in [Splunk](https://splunk.edx.org/en-US/app/search/search?q=search%20index%3D%22stage-edx%22%20%22striptags%22%20%22raised%20unexpected%3A%20AttributeError%22%20%22service_variant%3Dlms%22&earliest=-15m&latest=now&display.page.search.mode=smart&dispatch.sample_ratio=1&sid=1550125632.2982431)


## Standbox
Repeat the same steps for Stage
* Answer this problem (https://awaisdar001.sandbox.edx.org/courses/course-v1:edX+Test101+course/courseware/d7b00afc151e46c5ba1e370c310db679/789f0a355a09465b8d7210a61e1c229f/1?activate_block_id=block-v1%3AedX%2BTest101%2Bcourse%2Btype%40vertical%2Bblock%405b2f889da9a5421d970331a0ffe0d690)
* Go to Instructor Dashboard (https://awaisdar001.sandbox.edx.org/courses/course-v1:ArbiX+CS101+2015_T2/instructor#view-data_download)
* Select Problem (block-v1:ArbiX+CS101+2015_T2+type@problem+block@1d143ff22cdf4e68bb50bf749e9484a2)
* Click `Download a CSV of problem responses`
* Verify CSV generated successfully.

awaisdar001 commented:
jenkins run bokchoy

awaisdar001 commented:
jenkins run bokchoy

awaisdar001 commented:
Blocked by : https://github.com/edx/edx-platform/pull/19770

awaisdar001 commented on common/lib/capa/capa/tests/test_capa_problem.py:
```diff
@@ -467,16 +469,21 @@ def capa_problem(self, xml):
 
         Arguments:
-            problme_html (str): problem HTML
+            problem_html (str): problem HTML
```
just fixed a typo

DawoudSheraz commented on common/lib/capa/capa/tests/test_capa_problem.py:
```diff
@@ -467,16 +469,21 @@ def capa_problem(self, xml):
-    def assert_problem_html(self, problme_html, group_label, *input_labels):
+    def assert_problem_data(self, problem_data):
+        """Verify problem data is in expected state"""
+        for __, problem_value in six.viewitems(problem_data):
```
https://six.readthedocs.io/#six.viewvalues
six.viewvalues is another option.

Rabia23 commented:
@awaisdar001 do you have a sandbox to test the fix?

awaisdar001 commented:
> @awaisdar001 do you have a sandbox to test the fix?

Sure, I have updated the description of the PR and added links and instructions how to test on both stage and standbox.

awaisdar001 commented on common/lib/capa/capa/tests/test_capa_problem.py:
```diff
@@ -467,16 +469,21 @@ def capa_problem(self, xml):
-    def assert_problem_html(self, problme_html, group_label, *input_labels):
+    def assert_problem_data(self, problem_data):
+        """Verify problem data is in expected state"""
+        for problem_value in six.viewvalues(problem_data):
```
@DawoudSheraz Updated.

awaisdar001 commented:
jenkins run bokchoy
